### ISSUE TITLE
OmniWeaving new HY video support

### ISSUE THREAD

**Issue by zwukong** [Comfy-Org/ComfyUI#13268]
### Feature Idea

https://huggingface.co/tencent/HY-OmniWeaving

**Comment by MeiYi-dev** [comment]
https://github.com/Comfy-Org/ComfyUI/pull/13289 Someone has made a PR, you can test it out and report if any problems occur in that PR, so it could be merged.

**Comment by zwukong** [comment]
not a full implement i think, R2V is the most useful and important

**Comment by gitmylo** [comment]
> not a full implement i think, R2V is the most useful and important

Wdym? The pr implements `"t2v", "i2v", "interpolation", "reference2v", "editing", "tiv2v"` with the `HunyuanVideo15OmniConditioning` node. Seems like a full implementation for that.

**Comment by garrettashcroft1231-max** [comment]
I’d like to report a UI issue related to the prompt editing workflow

**Comment by dylanpowell1231-alt** [comment]
Thanks for reporting this. Could you provide a bit more detail about the UI
issue you’re encountering in the prompt editing workflow? Steps to
reproduce or screenshots would help a lot.

**Comment by zwukong** [comment]
"t2v", "i2v", "interpolation", "reference2v", "editing", "tiv2v"，really？ it will be great if author can provide a video demo each

**Comment by gitmylo** [comment]
I have tried out t2v and reference2v so far. I did notice that reference2v has a tendency to overlay the reference image in the first few frames, at least if you're using one reference image only.

Edit: The flashing image in reference2v was because I wasn't using clip vision outputs, using them fixes this. The PR doesn't seem to mention it, but you should use `sigclip_vision_patch14_384` as the clip vision model, it appears to be the correct and compatible option.

**Comment by SyntaxDrift42** [comment]
Thanks for the PR, works in my limited testing.  
This is a "vibe-coded" OmniWeaving LORA loader I made that seems to work properly for the files created by the official training script.  I claim little expertise; offered in case it's of use.  I left in an alpha widget setting since their script has that argument, one can set their model scaling.

It shows logging like:
```
[OmniWeaving LoRA] Loading: omni/pytorch_lora_weights.safetensors  (1564 keys)
[OmniWeaving LoRA] Converted 1534 keys → 1240 native keys
[OmniWeaving LoRA] Key map: 2367 entries
[OmniWeaving LoRA] Applied 444 patches
```

machine output:
Design intent: Fuse separate Q/K/V LoRA tensors into merged QKV format so PEFT-trained LoRAs work with ComfyUI's HunyuanVideo.
FLUX debt: FLUX's loader solved the identical problem—diffusers separate to_q/k/v vs native fused qkv—so I adapted their block-diagonal fusion math (A_fused = cat([A_q,A_k,A_v]), B_fused = block_diag(B_q,B_k,B_v)) and key normalization patterns.

(edit - replacing this script with a working copy.  seems my last minute effort to clean it up before posting actually broke it somehow.  I convinced it to work again, to some degree)
```py
import torch
import comfy.utils
import comfy.lora
import folder_paths
import logging

logger = logging.getLogger(__name__)

N_DOUBLE = 54


class OmniWeavingLoraLoader:
    @classmethod
    def INPUT_TYPES(cls):
        return {
            "required": {
                "model": ("MODEL",),
                "lora_name": (folder_paths.get_filename_list("loras"),),
                "strength": (
                    "FLOAT",
                    {
                        "default": 1.0,
                        "min": -10.0,
                        "max": 10.0,
                        "step": 0.01,
                    },
                ),
                "alpha": (
                    "FLOAT",
                    {
                        "default": 16.0,
                        "min": 0.1,
                        "max": 128.0,
                        "step": 0.5,
                    },
                ),
            },
            "optional": {
                "lora_name_override": ("STRING", {"forceInput": True}),
            },
        }

    RETURN_TYPES = ("MODEL",)
    FUNCTION = "load_lora"
    CATEGORY = "loaders/OmniWeaving"
    TITLE = "OmniWeaving LoRA Loader"

    def load_lora(self, model, lora_name, strength, alpha=16.0, lora_name_override=""):
        if strength == 0:
            return (model,)

        if lora_name_override and lora_name_override.strip():
            lora_name = lora_name_override.strip()

        lora_path = folder_paths.get_full_path("loras", lora_name)
        lora_sd = comfy.utils.load_torch_file(lora_path, safe_load=True)
        logger.info(f"[OmniWeaving LoRA] Loading: {lora_name}  ({len(lora_sd)} keys)")

        lora_sd = self._convert_to_native(lora_sd, alpha)

        key_map = self._build_key_map(model)
        patch_dict = comfy.lora.load_lora(lora_sd, key_map, log_missing=False)
        logger.info(f"[OmniWeaving LoRA] Applied {len(patch_dict)} patches")

        model_lora = model.clone()
        model_lora.add_patches(patch_dict, strength_patch=strength, strength_model=1.0)

        return (model_lora,)

    @staticmethod
    def _convert_to_native(lora_sd, alpha=16.0):
        norm = OmniWeavingLoraLoader._normalize_keys(lora_sd)
        out = {}
        done = set()

        for i in range(N_DOUBLE):
            db = f"double_blocks.{i}"

            OmniWeavingLoraLoader._fuse_qkv(
                norm,
                out,
                done,
                q=f"{db}.img_attn_q",
                k=f"{db}.img_attn_k",
                v=f"{db}.img_attn_v",
                dst=f"diffusion_model.{db}.img_attn.qkv",
                alpha=alpha,
            )

            OmniWeavingLoraLoader._fuse_qkv(
                norm,
                out,
                done,
                q=f"{db}.txt_attn_q",
                k=f"{db}.txt_attn_k",
                v=f"{db}.txt_attn_v",
                dst=f"diffusion_model.{db}.txt_attn.qkv",
                alpha=alpha,
            )

            for src, dst in [
                (f"{db}.img_attn_proj", f"diffusion_model.{db}.img_attn.proj"),
                (f"{db}.txt_attn_proj", f"diffusion_model.{db}.txt_attn.proj"),
                (f"{db}.img_mlp.fc1", f"diffusion_model.{db}.img_mlp.0"),
                (f"{db}.img_mlp.fc2", f"diffusion_model.{db}.img_mlp.2"),
                (f"{db}.txt_mlp.fc1", f"diffusion_model.{db}.txt_mlp.0"),
                (f"{db}.txt_mlp.fc2", f"diffusion_model.{db}.txt_mlp.2"),
                (f"{db}.img_mod.linear", f"diffusion_model.{db}.img_mod.linear"),
                (f"{db}.txt_mod.linear", f"diffusion_model.{db}.txt_mod.linear"),
            ]:
                OmniWeavingLoraLoader._remap(norm, out, done, src, dst)

        for src, dst in [
            ("mm_in.linear_1", "diffusion_model.mm_in.linear_1"),
            ("mm_in.linear_2", "diffusion_model.mm_in.linear_2"),
            ("time_in.mlp.0", "diffusion_model.time_in.mlp.0"),
            ("time_in.mlp.2", "diffusion_model.time_in.mlp.2"),
            (
                "txt_in.c_embedder.linear_1",
                "diffusion_model.txt_in.c_embedder.linear_1",
            ),
            (
                "txt_in.c_embedder.linear_2",
                "diffusion_model.txt_in.c_embedder.linear_2",
            ),
            ("txt_in.t_embedder.mlp.0", "diffusion_model.txt_in.t_embedder.in_layer"),
            ("txt_in.t_embedder.mlp.2", "diffusion_model.txt_in.t_embedder.out_layer"),
            ("byt5_in.fc1", "diffusion_model.byt5_in.fc1"),
            ("byt5_in.fc2", "diffusion_model.byt5_in.fc2"),
            ("byt5_in.fc3", "diffusion_model.byt5_in.fc3"),
        ]:
            OmniWeavingLoraLoader._remap(norm, out, done, src, dst)

        for key, val in norm.items():
            if key not in done:
                out[key] = val

        n_converted = sum(1 for k in done if k in norm)
        logger.info(
            f"[OmniWeaving LoRA] Converted {n_converted} keys → {len(out)} native keys"
        )
        return out

    @staticmethod
    def _normalize_keys(lora_sd):
        out = {}
        for key, val in lora_sd.items():
            k = key
            for pfx in ("transformer.", "diffusion_model.", "unet.", ""):
                if k.startswith(pfx):
                    k = k[len(pfx) :]
                    break
            k = k.replace(".lora_down.", ".lora_A.").replace(".lora_up.", ".lora_B.")
            out[k] = val
        return out

    @staticmethod
    def _alpha_scale(norm, base, alpha_override=None):
        down_key = f"{base}.lora_A.weight"
        alpha_key = f"{base}.alpha"
        if alpha_override is not None:
            rank = norm[down_key].shape[0] if down_key in norm else 1
            return alpha_override / rank if rank > 0 else 1.0
        if alpha_key in norm and down_key in norm:
            rank = norm[down_key].shape[0]
            if rank > 0:
                return float(norm[alpha_key]) / rank
        return 1.0

    @staticmethod
    def _fuse_qkv(norm, out, done, q, k, v, dst, alpha=None):
        keys_A = [f"{b}.lora_A.weight" for b in (q, k, v)]
        keys_B = [f"{b}.lora_B.weight" for b in (q, k, v)]

        if not all(kk in norm for kk in keys_A + keys_B):
            return

        A_q, A_k, A_v = [norm[kk] for kk in keys_A]
        B_q, B_k, B_v = (
            norm[kk] * OmniWeavingLoraLoader._alpha_scale(norm, b, alpha)
            for kk, b in zip(keys_B, (q, k, v))
        )

        r_q, r_k, r_v = A_q.shape[0], A_k.shape[0], A_v.shape[0]
        o_q, o_k, o_v = B_q.shape[0], B_k.shape[0], B_v.shape[0]
        r_total = r_q + r_k + r_v
        o_total = o_q + o_k + o_v

        A_fused = torch.cat([A_q, A_k, A_v], dim=0)

        B_fused = torch.zeros(o_total, r_total, dtype=B_q.dtype, device=B_q.device)
        B_fused[0:o_q, 0:r_q] = B_q
        B_fused[o_q : o_q + o_k, r_q : r_q + r_k] = B_k
        B_fused[o_q + o_k : o_total, r_q + r_k : r_total] = B_v

        out[f"{dst}.lora_A.weight"] = A_fused
        out[f"{dst}.lora_B.weight"] = B_fused
        out[f"{dst}.alpha"] = torch.tensor(float(r_total))

        for kk in keys_A + keys_B:
            done.add(kk)
        done.add(f"{q}.alpha")
        done.add(f"{k}.alpha")
        done.add(f"{v}.alpha")

    @staticmethod
    def _remap(norm, out, done, src_base, dst_base):
        key_A = f"{src_base}.lora_A.weight"
        key_B = f"{src_base}.lora_B.weight"
        if key_A not in norm or key_B not in norm:
            return
        out[f"{dst_base}.lora_A.weight"] = norm[key_A]
        out[f"{dst_base}.lora_B.weight"] = norm[key_B]
        done.update([key_A, key_B])
        alpha_key = f"{src_base}.alpha"
        if alpha_key in norm:
            out[f"{dst_base}.alpha"] = norm[alpha_key]
            done.add(alpha_key)

    @staticmethod
    def _build_key_map(model):
        key_map = {}
        for model_key in model.model.state_dict().keys():
            if not model_key.endswith(".weight"):
                continue

            base = model_key[: -len(".weight")]
            bare = (
                base[len("diffusion_model.") :]
                if base.startswith("diffusion_model.")
                else base
            )

            for pfx in ("diffusion_model.", "transformer.", ""):
                key_map[f"{pfx}{bare}"] = model_key

        logger.info(f"[OmniWeaving LoRA] Key map: {len(key_map)} entries")
        return key_map


NODE_CLASS_MAPPINGS = {
    "OmniWeavingLoraLoader": OmniWeavingLoraLoader,
}

NODE_DISPLAY_NAME_MAPPINGS = {
    "OmniWeavingLoraLoader": "OmniWeaving LoRA Loader",
}

__all__ = ["NODE_CLASS_MAPPINGS", "NODE_DISPLAY_NAME_MAPPINGS"]
```
